### PR TITLE
Initial functionality provided for action items for improving records

### PR DESCRIPTION
**Pull request by emredalkiran**
Provides basic functionality for action items stated in #5 

It has some styling issues, but before working on it anymore, I wanted to check if the current approach is reasonable.

### CODE REVIEW

File: src/components/records.vue [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      clipboard.writeText(JSON.stringify(record, null, 2))
+    }
+  }
+}
+</script>
+
+<template lang="pug">
+  div.record-root
+    modal(v-if="isModalVisible" @close="isModalVisible = false" :record="modalRecord")
+    record.asd(v-for="record in records" :record="record" :tabindex="0" :rootname="'Document'" :indentLevel="1" v-on:dblclick.native ="isModalVisible = !isModalVisible; modalRecord = record; " @keydown.native.meta.8="$emit('deleteRecord', record)" @keydown.native.meta.67="copyToClipboard(record)")
```
Comment (dashersw) [inline src/components/records.vue:26]: are you sure about `.asd`? :)

Comment (dashersw) [reply src/components/records.vue:26]: how do we trigger delete and copy?

**Comment by dashersw** [comment]
This is fantastic work @emredalkiran, thank you!

File: src/components/modal.vue [hunk 2]
```diff
@@ -0,0 +1,92 @@
+<script>
+export default {
+  name: 'modal',
+  data() {
```
Comment (dashersw) [inline src/components/modal.vue:4]: we should remove unused properties, including the `methods` object.

File: src/components/modal.vue [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+</script>
+<template lang="pug"> 
+  transition
+      div.modal-mask
+        div.modal-wrapper
+          div.modal-container
+            div.modal-body 
+              textarea(:spellcheck="false") {{ record | pretty }}
+            div.modal-footer
+              button.modal-default-button(@click="$emit('close')") OK
```
Comment (dashersw) [inline src/components/modal.vue:24]: would be great to capture escape to close the modal.

File: src/store/index.js [hunk 4]
```diff
@@ -13,7 +13,8 @@ export default new Vuex.Store({
     collections: [],
     records: [],
     collectionName: '',
-    totalNumberOfResults: null
+    totalNumberOfResults: null,
+    recordList: []
```
Comment (dashersw) [inline src/store/index.js:17]: what is recordList?

File: src/components/records.vue (hunk 1 above)
Comment (emredalkiran) [reply src/components/records.vue:26]: I have no idea about asd :) Probably, was trying about something about styling and forgot to clean up my mess. Delete and copy should be working as you specified in issue details.

File: src/store/index.js (hunk 4 above)
Comment (emredalkiran) [reply src/store/index.js:17]: I was trying a clean way to represent nested documents recursively. Again, forgot to clean up afterwards.

**Comment by emredalkiran** [comment]
@dashersw thanks for the feedback and kind words. record.vue has some problems, it doesn't repsresent _id key properly, by adding additional nesting of id in the form of an array. On top of that, if you delete all the documents on a query result or left a few documents in the collapsed mode, the alternating background also gets removed because of the wrapping html element's size has shrunken (in order to support scrolling and keeping alternate between collapsed/open rows, I alternated the background color of the wrapper element).  The border of the records view remains same though, which creates an awkward view. I'll try to address these problems as well as the ones you mentioned above.

In addition to that, it is not very clear which document has focus for deleting. I think adding some sort of highlighting for active document would also be beneficial.

I think font size for record displaying can be decreased a little bit and color of the records text can be darkened a little, towards a greyish white, considering dark background color.

**Comment by dashersw** [comment]
Thank you! I agree. I think overall the font sizes are too big in the project :) we should decrease them.

**Pull request closed without merging**